files/default/plugins/printers.rb

```diff
@@ -0,0 +1,44 @@
+provides 'ohai_gecos'
+
+if ohai_gecos.nil?
+  ohai_gecos Mash.new
+end
+
+begin
+  entries = []
+  printers = []
+
+  entries = Mixlib::ShellOut.new("lpstat -a | awk {'print $1'}")
+  entries.run_command
+
+  if entries.exitstatus == 0
+
+    entries.stdout.split(/\r?\n/).each do |printer|
+      lpoptions = Mixlib::ShellOut.new("lpoptions -p #{printer}")
+      lpoptions.run_command
+      lpq = Mixlib::ShellOut.new("lpq -P #{printer}")
+      lpq.run_command
+      lpstat = Mixlib::ShellOut.new("lpstat -p #{printer}")
+      lpstat.run_command
+
```

Comment:
Even if valid printer names are enforced in line 11, exit status of lp* commands should be checked for non error exit status before writing data to the node.